pyrobosim/pyrobosim/utils/pose.py

```diff
@@ -3,26 +3,54 @@
 """
 
 import numpy as np
-
+from transforms3d.euler import euler2quat, quat2euler
+from transforms3d.quaternions import qnorm
 
 class Pose:
-    """ Represents a 2.5D (X, Y, Z, yaw) pose. """
+    """ Represents a 3D pose. """
 
-    def __init__(self, x=0.0, y=0.0, z=0.0, yaw=0.0):
-        """ Creates a new Pose object. """
+    def __init__(self, x=0.0, y=0.0, z=0.0,
+                 roll=0.0, pitch=0.0, yaw=0.0,
+                 q=None):
+        """
+        Creates a new Pose object.
+        
+        :param x: X position
+        :type x: float
+        :param y: Y position
+        :type y: float
+        :param z: Z position
+        :type z: float
+        :param roll: Roll angle (about X axis), in radians
+        :type roll: float
+        :param pitch: Pitch angle (about Y axis), in radians
+        :type pitch: float
+        :param yaw: Yaw angle (about Z axis), in radians
+        :type yaw: float
+        :param q: Quaternion, specified at [qw, qx, qy, qz] 
+            If specified, will override roll/pitch/yaw values.
+        :type q: list[float]
+        """
         self.x = x
         self.y = y
         self.z = z
-        self.yaw = wrap_angle(yaw)
+
+        if q is not None:
+            self.set_quaternion(q)
+        else:
+            self.set_euler_angles(roll, pitch, yaw)
 
     @classmethod
     def from_list(cls, plist):
         """ 
         Creates a pose from a list. The assumptions are:
         
         * 2-element lists: ``[x, y]``
-        * 3-element lists: ``[x, y, yaw]``
+        * 3-element lists: ``[x, y, z]``
```

Comment:
Still not sure which 3-element definition I prefer... maybe [x, y, z] is the way to go

Reply:
Actually this change broke YAML file loading which was expecting [x, y, yaw] so I had to update that to use [x, y, z, yaw] for now.